File: sdk/csharp/driversdk/DriverSDK/DmContext/Models/AccessTemplate.cs
using System.Xml.Linq;

namespace DriverSDK.DmContext.Models
{
    public class AccessTemplate
    {
        public string Name { get; set; } = "";
        public string Version { get; set; } = "";
        public IList<DeviceProperty>? Properties { get; set; }
        public IList<ModelMapping>? Mappings { get; set; }

        public AccessTemplate() { }

        public AccessTemplate(string name, string version, IList<DeviceProperty> properties,  IList<ModelMapping> mappings)
        {
            this.Name = name;
            this.Version = version;
            this.Properties = properties;
            this.Mappings = mappings;
        }
    }

    public class AccessConfig
    {
        public string Custom { get; set; } = "";

        public AccessConfig() { }

        public AccessConfig(string custom)
        {
            this.Custom = custom;
        }
    }

    public class ModelMapping
    {
        public string Attribute { get; set; } = "";
        public string Type { get; set; } = "";
        public string Expression { get; set; } = "";
        public int Precision { get; set; } = 0;
        public double Deviation { get; set; } = 0;
        public int SilentWin { get; set; } = 0;

        public ModelMapping() { }

        public ModelMapping(string attribute, string type, string expression, int precision, double deviation, int silentWin) 
        {
            this.Attribute = attribute;
            this.Type = type;
            this.Expression = expression;
            this.Precision = precision;
            this.Deviation = deviation;
            this.SilentWin = silentWin;
        }
    }
}

File: sdk/cxx/driversdk/src/models/accesstemplate.cpp
#include "accesstemplate.h"

namespace DRIVERSDK {

// Default constructor
    AccessTemplate::AccessTemplate() {}

// Parameterized constructor
    AccessTemplate::AccessTemplate(const std::string& templateName, const std::string& templateVersion,
                                   const std::vector<DeviceProperty>& templateProperties, const std::vector<ModelMapping>& templateMappings)
            : name(templateName), version(templateVersion), properties(templateProperties), mappings(templateMappings) {}

// Getter and setter methods
    const std::string& AccessTemplate::getName() const {
        return name;
    }

    void AccessTemplate::setName(const std::string& templateName) {
        name = templateName;
    }

    const std::string& AccessTemplate::getVersion() const {
        return version;
    }

    void AccessTemplate::setVersion(const std::string& templateVersion) {
        version = templateVersion;
    }

    const std::vector<DeviceProperty>& AccessTemplate::getProperties() const {
        return properties;
    }

    void AccessTemplate::setProperties(const std::vector<DeviceProperty>& templateProperties) {
        properties = templateProperties;
    }

    const std::vector<ModelMapping>& AccessTemplate::getMappings() const {
        return mappings;
    }

    void AccessTemplate::setMappings(const std::vector<ModelMapping>& templateMappings) {
        mappings = templateMappings;
    }

} // namespace DRIVERSDK
